### ISSUE TITLE
Cannot get the webgui to run

### ISSUE THREAD

**Issue by GordonApplepie** [ph1p/ikea-led-obegraensad#103]
Hi there,
hope u guys can help me. I do have multiple questions.

1. Can I run the webgui without being connected to the IKEA matrix? (I wanted to do software stuff first before I break open the matrix and soldier out the old controller)
2. I wasn't aware that the differences on ESP32 controllers are that big when I ordered my one. I got a **ESP32 NodeMCU Module USB C ESP32 WROOM 32 Nodemcu Development Board 2.4 GHz WLAN WiFi Bluetooth CH340 Chip** from Amazon.

I basically uploaded the standard configuration, connected through Wifi and setup the AP settings. Which worked fine, then fixed the IP in my router. When I try to ping or open the IP in the web browser I am not able to reach it though.

So I tried to setup the IP also in code, for example in **frontend/.env** (which should only matter when actually making a request I think). Then I tried setting up the static ip config in **constants.h**. But it will still not work.

I also tried to debug but then I get an error that states that there is not enough flash memory for it.

Also even after uploading an esp32 image again the AP settings seem to be stored. How am I able to delete them?

**Comment by jaal2001** [comment]
No. 1 yes, you can. The display is "just the output device". Connect your ESP32 to USB Power (after flashing it) and you are good to go.
No. 2 As far as I can see, your ESP should work fine. You are also telling, that you could successfully connect to the ESP after flashing, so it works (which already had solved question No. 1)

What do you mean by "then fixed the IP in my router"?

**Comment by GordonApplepie** [comment]
In my router I enabled a fixed IP for the board. That is what I meant.
I got a little further. So I installed Teleplot and found out that there was an mDNS error which I could fix with these lines in **main.cpp**
![image](https://github.com/ph1p/ikea-led-obegraensad/assets/18107548/affdd4f2-a00c-4c10-9f12-eaae36aa5005)

Now the output in Teleplot looks actually good. But I am not able to open up the gui on the IP in my web browser 🤔 
![image](https://github.com/ph1p/ikea-led-obegraensad/assets/18107548/4873ddd3-2beb-4af0-b73f-0aad0a47c84b)

**Comment by jaal2001** [comment]
Okay, that is beyond where I could help. My suggestion would be: Try DHCP first, check if the board then works fine (with the non modified code). If that works and you can access the board, maybe you can proceed from there.

**Comment by GordonApplepie** [comment]
ok. it is getting weirder 🤔 so i added a httpclient to actually request the gui from the esp32 within and it is actually working
![image](https://github.com/ph1p/ikea-led-obegraensad/assets/18107548/32c5bfb6-f07a-4dde-8d6b-716ec9973c5d)
![image](https://github.com/ph1p/ikea-led-obegraensad/assets/18107548/2b5bda1b-8957-46ee-91fe-17f528b6fd5f)

this makes no sense to me. i could only image there is some kind of firewall inbetween, but there is none 👀 same wifi ap

**Comment by GordonApplepie** [comment]
so i also tried setting up static ip. but no change 🤔 mh. no idea what can be the issue
![image](https://github.com/ph1p/ikea-led-obegraensad/assets/18107548/eb540693-a14e-45fa-8b70-9739aa276687)
![image](https://github.com/ph1p/ikea-led-obegraensad/assets/18107548/14982dc9-8429-4739-9d3f-e92e95d89d7b)

**Comment by jaal2001** [comment]
Any chance to use that ESP on an another network (i.e. Hotspot of mobile)? This could maybe help to get better into the problem.

**Comment by ph1p** [comment]
I will close this. Feel free to open it again :)
